src/testrel.jl

```diff
@@ -414,19 +414,57 @@ function _test_rel_steps(;
     test_engine = user_engine === nothing ? get_test_engine() : user_engine
     @debug("$name: using test engine: $test_engine")
 
+    logger = TestLogger()
+    logged = false
+
     try
         type = quiet ? QuietTestSet : Test.DefaultTestSet
-        @testset type "$(string(name))" begin
-            create_test_database(schema, clone_db)
-            elapsed_time = @timed begin
-                for (index, step) in enumerate(steps)
-                    _test_rel_step(index, step, schema, test_engine, name, length(steps))
+        ts = Logging.with_logger(logger) do 
+            @testset type "$(string(name))" begin
+                create_test_database(schema, clone_db)
+                elapsed_time = @timed begin
+                    for (index, step) in enumerate(steps)
+                        _test_rel_step(index, step, schema, test_engine, name, length(steps))
+                    end
+                end
+                stats = (time = elapsed_time.time, allocations = elapsed_time.gcstats.poolalloc, bytes = elapsed_time.gcstats.allocd)
+                @info("$name: $stats")
+            end
+        end
+
+        # Print summary of logs from the testset, or all logs if it did not pass
+        if anynonpass(ts)
+            # dump all of the captured logs
+            io = IOBuffer()
+            write(io, "Test $name failed\n\n CAPTURED LOGS:\n")
+            redirect_stdio(stdout=io, stderr=io) do
+                playback_log.(logger.logs)
+            end
+            msg = String(take!(io))
+            @warn msg database=schema engine_name=test_engine test_name=name passed=false
+        else
```

Comment:
should this respect quiet?

Reply:
`quiet` could be more descriptive. It quiets the summary output from the testrel testset. This is needed since when used with the concurrent testset it thinks it's the parent. This is unrelated to output from the test itself.

So, no, don't respect `quiet`.

Reply:
Standard testsets output a summary, unless they are nested in another testset in which case they pass on the data for the parent to print or pass on.

ConcurrentTestSet breaks this and uses a reference to fetch the results from the children testsets. The children then think they are the outermost parent. QuietTestSet exists to stop the summary output and just pass the results on when asked.

I was attempting to work alongside `@testset` and maintain compatibility, but in retrospect I think it would have been better to use `RAITest.@testset` for nested concurrent testsets.

Reply:
Thanks for the context! So is it okay to use the `QuietTestSet` for all these `@test_rel` desugarings? Or does this cause weirdsies in the case of a `@test_rel` not in a testset?

I'm guessing the latter, so I'll play around with this a bit and maybe rename/refactor some things.